TransferMethodRepository/Tests/RemoteTransferMethodConfigurationRepositoryTests.swift

```diff
@@ -0,0 +1,335 @@
+    private let currency = "USD"
+    private let transferMethodType = "BANK_ACCOUNT"
+    private let profileType = "INDIVIDUAL"
+    private lazy var fieldsResponseData = HyperwalletTestHelper
+        .getDataFromJson("TransferMethodConfigurationFieldsResponse")
+    private lazy var keyResponseData = HyperwalletTestHelper.getDataFromJson("TransferMethodConfigurationKeysResponse")
+    private static var userResponseData = HyperwalletTestHelper.getDataFromJson("UserIndividualResponse")
+
+    override func setUp() {
+        Hyperwallet.setup(HyperwalletTestHelper.authenticationProvider)
+    }
+
+    override func tearDown() {
+        if Hippolyte.shared.isStarted {
+            Hippolyte.shared.stop()
+        }
+    }
+
+    func testGetKeys_success() {
+        RemoteTransferMethodConfigurationRepositoryTests
+            .setupResponseMockServer(keyResponseData)
+        let expectation = self.expectation(description: "Get transfer methods keys")
+        var transferMethodConfigurationKey: HyperwalletTransferMethodConfigurationKey?
+        var error: HyperwalletErrorType?
+        let repository = RemoteTransferMethodConfigurationRepository()
+
+        // When
+        repository.getKeys { (result) in
+            switch result {
+            case .success(let resultKey):
+                transferMethodConfigurationKey = resultKey
+
+            case .failure(let resultError):
+                error = resultError
+            }
+            expectation.fulfill()
+        }
+        wait(for: [expectation], timeout: 1)
+
+        XCTAssertNil(error, "The error should be nil")
+        XCTAssertNotNil(transferMethodConfigurationKey, "The result should not be nil")
+        XCTAssertGreaterThan(transferMethodConfigurationKey!.countries()!.count, 0)
+        XCTAssertGreaterThan(transferMethodConfigurationKey!.currencies(from: country)!.count, 0)
+    }
+
+    func testGetKeys_failureWithError() {
+        RemoteTransferMethodConfigurationRepositoryTests
+            .setupResponseMockServer(keyResponseData, NSError(domain: "", code: -1009, userInfo: nil))
+        let expectation = self.expectation(description: "Get transfer methods keys")
+        var transferMethodConfigurationKey: HyperwalletTransferMethodConfigurationKey?
+        var error: HyperwalletErrorType?
+        let repository = RemoteTransferMethodConfigurationRepository()
+
+        // When
+        repository.getKeys { (result) in
+            switch result {
+            case .success(let resultKey):
+                transferMethodConfigurationKey = resultKey
+
+            case .failure(let resultError):
+                error = resultError
+            }
+            expectation.fulfill()
+        }
+        wait(for: [expectation], timeout: 1)
+
+        XCTAssertNotNil(error, "The error should not be nil")
+        XCTAssertNil(transferMethodConfigurationKey, "The result should be nil")
+    }
+
+    func testGetKeys_successWithKeyResultWhenNotNil() {
```

Comment:
If it's from cache then maybe rename it to testGetKeys_successWithKeyResultFromCache